### PR TITLE
feat!: updates for common config

### PR DESCRIPTION
**Pull request by ejlee3**
BREAKING CHANGE: configuration file changed to remove common config settings

Signed-off-by: Elizabeth J Lee <elizabeth.j.lee@intel.com>

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-virtual-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) - updates for common config bootstrapping
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) - docs need to be added for common config
  <link to docs PR>

## Testing Instructions

1. run edgex non-secure stack
2. stop the device virtual container
3. build device virtual - `make build`
4. run device virtual using config provider
5. verify that private configuration is loaded in consul
6. ensure that writable in the private config is modifyable - change log level and see change in the terminal

### CODE REVIEW

File: cmd/res/configuration.toml [hunk 1]
```diff
@@ -2,106 +2,32 @@ MaxEventSize = 0 # value 0 unlimit the maximum event size that can be sent to me
...
-      password = ""
   [Writable.Reading]
   ReadingUnits = true
   [Writable.Telemetry]
-  Interval = "30s"
-  PublishTopicPrefix  = "edgex/telemetry" # /<service-name>/<metric-name> will be added to this Publish Topic prefix
     [Writable.Telemetry.Metrics] # All service's metric names must be present in this list.
     # Device SDK provided Metrics
     EventsSent = false
     ReadingsSent = false
```
Comment (lenny-intel) [inline cmd/res/configuration.toml:11]: This is in Device Services Common. I.e no service specific metrics.

File: cmd/res/configuration.toml [hunk 2]
```diff
@@ -2,106 +2,32 @@ MaxEventSize = 0 # value 0 unlimit the maximum event size that can be sent to me
...
-[Registry]
-Host = "localhost"
-Port = 8500
-Type = "consul"
 
 [Clients]
   [Clients.core-data]
   Protocol = "http"
   Host = "localhost"
   Port = 59880
```
Comment (lenny-intel) [inline cmd/res/configuration.toml:22]: No longer used with recent PRs.

File: cmd/res/configuration.toml [hunk 3]
```diff
@@ -2,106 +2,32 @@ MaxEventSize = 0 # value 0 unlimit the maximum event size that can be sent to me
...
 LogLevel = "INFO"
-  # InsecureSecrets are required for when Redis is used for message bus
-  [Writable.InsecureSecrets]
-    [Writable.InsecureSecrets.DB]
-    path = "redisdb"
-      [Writable.InsecureSecrets.DB.Secrets]
-      username = ""
-      password = ""
   [Writable.Reading]
   ReadingUnits = true
```
[the remark this answers is not in the document]
Comment (ejlee3) [reply cmd/res/configuration.toml:6]: this is already in DeviceServices common - I missed it.

File: cmd/res/configuration.toml (hunk 1 above)
Comment (ejlee3) [reply cmd/res/configuration.toml:11]: ok. removing them

File: cmd/res/configuration.toml [hunk 4]
```diff
@@ -2,106 +2,19 @@ MaxEventSize = 0 # value 0 unlimit the maximum event size that can be sent to me
...
-DevicesDir = "./res/devices"
-UpdateLastConnected = false
-EnableAsyncReadings = true
-AsyncBufferSize = 16
-Labels = []
-UseMessageBus = true
-  [Device.Discovery]
-  Enabled = false
-  Interval = "30s"
+  ProvisionWatchersDir = ""
```
Comment (lenny-intel) [inline cmd/res/configuration.toml:20]: No longer needed with the tweak we made to common.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: cmd/res/configuration.toml (hunk 4 above)
Comment (ejlee3) [reply cmd/res/configuration.toml]: fixed

**Review by lenny-intel — DISMISSED** [review]
LGTM

**Comment by cloudxxx8** [comment]
recheck

**Comment by cloudxxx8** [comment]
something wrong in the unit test
may not to upgrade contract version for now

File: go.mod [hunk 5]
```diff
@@ -3,8 +3,8 @@ module github.com/edgexfoundry/device-virtual-go
 go 1.18
 
 require (
-	github.com/edgexfoundry/device-sdk-go/v3 v3.0.0-dev.22
-	github.com/edgexfoundry/go-mod-core-contracts/v3 v3.0.0-dev.10
+	github.com/edgexfoundry/device-sdk-go/v3 v3.0.0-dev.26
+	github.com/edgexfoundry/go-mod-core-contracts/v3 v3.0.0-dev.14
```
Comment (lenny-intel) [inline go.mod:7]: Needs to be 
```suggestion
	github.com/edgexfoundry/go-mod-core-contracts/v3 v3.0.0-dev.10
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**